File: redsheet-portal/src/app/database-models/password-policy.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class PasswordPolicy {
    constructor(passwordPolicy?: PasswordPolicy )  {
        let properties = [ "forcePasswordAlphaNum", "minimumNumberOfCharacters", "numberOfAttempts", "passwordExpiryDuration", "passwordPolicyId",];
        for (let property of properties)
            if (passwordPolicy && passwordPolicy[property])
                this[property] = passwordPolicy[property];
    }
 
	forcePasswordAlphaNum : boolean = false ;
 
    @range(1,2147483647)
	minimumNumberOfCharacters : number =   undefined;
 
    @range(1,2147483647)
	numberOfAttempts : number =   undefined;
 
    @range(1,2147483647)
	passwordExpiryDuration : number =   undefined;
 
	passwordPolicyId : number =   0 ;


}

File: RedSheet.DbEntities/Models/PasswordPolicy.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("PasswordPolicies", Schema = "dbo")]
	[Serializable]
	public partial class PasswordPolicy  
	{
 
	[Required]
	public bool ForcePasswordAlphaNum { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int MinimumNumberOfCharacters { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int NumberOfAttempts { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int PasswordExpiryDuration { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int PasswordPolicyId { get; set; } 

	public PasswordPolicy()
	{
    }

    }
}

